internal/server/acme/acme.go

```diff
@@ -102,148 +95,66 @@ func UpdateCertificate(s *state.State, challengeType string, provider ChallengeP
-			PrivateKey:  certInfo.PrivateKey(),
-		}, nil
-	}
-
-	// Generate new private key for user. This key needs to be different from the server's private key.
-	privateKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
-	if err != nil {
-		return nil, fmt.Errorf("Failed generating private key for user account: %w", err)
-	}
-
-	user := user{
-		Email: email,
-		Key:   privateKey,
-	}
-
-	config := lego.NewConfig(&user)
-
-	if caURL != "" {
-		config.CADirURL = caURL
-	} else {
-		// Default URL for Let's Encrypt
-		config.CADirURL = "https://acme-v02.api.letsencrypt.org/directory"
-	}
-
-	config.Certificate.KeyType = certcrypto.RSA2048
-
-	client, err := lego.NewClient(config)
-	if err != nil {
-		return nil, fmt.Errorf("Failed to create new client: %w", err)
-	}
-
-	err = provider.RegisterWithSolver(client.Challenge)
-	if err != nil {
-		return nil, fmt.Errorf("Failed setting challenge provider: %w", err)
 	}
 
-	var reg *registration.Resource
-
-	// Registration might fail randomly (as seen in manual tests), so retry in that case.
-	for i := 0; i < retries; i++ {
-		reg, err = client.Registration.Register(registration.RegisterOptions{TermsOfServiceAgreed: true})
-		if err == nil {
-			break
-		}
-
-		// In case we were rate limited, don't try again.
-		details, ok := err.(*acme.ProblemDetails)
-		if ok && details.Type == "urn:ietf:params:acme:error:rateLimited" {
-			break
-		}
-
-		l.Warn("Failed to register user, retrying in 10 seconds", logger.Ctx{"err": err})
-		time.Sleep(10 * time.Second)
-	}
+	args = append(args, "run")
 
+	_, _, err = subprocess.RunCommandSplit(context.TODO(), env, nil, "lego", args...)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to register user: %w", err)
-	}
-
-	user.Registration = reg
-
-	request := certificate.ObtainRequest{
-		Domains:    []string{domain},
-		Bundle:     true,
-		PrivateKey: certInfo.KeyPair().PrivateKey,
-	}
-
-	var certificates *certificate.Resource
-
-	l.Info("Issuing certificate")
-
-	// Get new certificate.
-	// This might fail randomly (as seen in manual tests), so retry in that case.
-	for i := 0; i < retries; i++ {
-		certificates, err = client.Certificate.Obtain(request)
-		if err == nil {
-			break
-		}
-
-		// In case we were rate limited, don't try again.
-		details, ok := err.(*acme.ProblemDetails)
-		if ok && details.Type == "urn:ietf:params:acme:error:rateLimited" {
-			break
-		}
-
-		l.Warn("Failed to obtain certificate, retrying in 10 seconds", logger.Ctx{"err": err})
-		time.Sleep(10 * time.Second)
+		return nil, fmt.Errorf("Failed to run lego command: %w", err)
 	}
 
+	certInfo, err = localtls.KeyPairAndCA(tmpDir+"/certificates", domain, localtls.CertServer, true)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to obtain certificate: %w", err)
+		return nil, fmt.Errorf("Failed to load certificate and key file: %w", err)
 	}
 
-	l.Info("Finished issuing certificate")
-
-	return certificates, nil
+	return &certificate.Resource{
```

Comment:
This is from `lego`, but is only used here for a convenient return type and to match the previous HTTP01Provider.